integrations/Satismeter/browser.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-var */
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable class-methods-use-this */
+import logger from "../../utils/logUtil";
+import { NAME } from "./constants";
+import { LOAD_ORIGIN } from "../ScriptLoader";
+
+class Satismeter {
+  constructor(config, analytics) {
+    logger.debug("in satismetwr const");
+    if (analytics.logLevel) {
+      logger.setLogLevel(analytics.logLevel);
+    }
+    this.name = NAME;
+    this.writeKey = config.writeKey;
+  }
+
+  init() {
+    logger.debug("===In init Satismeter===");
+    (function () {
+      var script = document.createElement("script");
+      var parent = document.getElementsByTagName("script")[0].parentNode;
+      window.satismeter =
+        window.satismeter ||
+        function () {
+          (window.satismeter.q = window.satismeter.q || []).push(arguments);
+        };
+      window.satismeter.l = 1 * new Date();
+      script.async = 1;
+      script.src = "https://app.satismeter.com/js";
+      script.setAttribute("data-loader", LOAD_ORIGIN);
+      parent.appendChild(script);
+    })();
+    window.satismeter({
+      writeKey: this.writeKey,
+    });
+  }
+
+  isLoaded() {
+    logger.debug("===In isLoaded Satismeter===");
+    return !!window.satismeter;
+  }
+
+  isReady() {
+    logger.debug("===In isReady Satismeter===");
+    return !!window.satismeter;
+  }
+
+  identify(rudderElement) {
+    logger.debug("===In Satismeter identify===");
+    const { message } = rudderElement;
+    const { traits } = message.context;
+    const userId = message.userId || traits?.userId;
```

Comment:
Shouldn't we also track the anonymous users also? Let's provide an option on the dashboard to track or not to track anonymous users and depending on that we can implement this.

Reply:
We are already tracking anonymous users in such a way that if track call is made without identify call then it will be anonymous user otherwise details could be taken from identify call

Reply:
Didn't get it. We are not providing any user traits through the track call.

Reply:
Actually while the loading the script we are sending a request with the writeKey which will be helping it to track a response even an anonymous one. But if we are making an identify call first and then track call then it captures the details from the identify call made previously through cookies and all.